### PR TITLE
Add ETag header.

### PR DESCRIPTION
**Pull request by Wosin**

### CODE REVIEW

**Comment by jdegoes** [comment]
@Wosin Can you please resolve conflicts?

File: zio-http/src/main/scala/zio/http/api/HeaderInputs.scala [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package zio.http.api
 
 import zio.http.model.HeaderNames
-import zio.http.model.headers.values.{Accept, Age, Allow, CacheControl, ContentLength, Origin}
-import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
```
Comment (ccerbusca) [inline zio-http/src/main/scala/zio/http/api/HeaderInputs.scala:5]: do not remove this import

Comment (Wosin) [reply zio-http/src/main/scala/zio/http/api/HeaderInputs.scala:5]: Uhhh, useless Intellij :D Fixed.

**Comment by jdegoes** [comment]
Hi @Wosin! As with the others, I will close until conflicts are fixed, feel free to open when ready to get this merged!

**Pull request closed without merging**